### PR TITLE
Turn-off Poisson solver if there is only 1 macroparticle (or less)

### LINKED ISSUE #44: Handling of Single Particles / Perfectly Flat / Line Beams
While running in debug (#43), I realized the `ResizeMesh` function does assert the simulation:
https://github.com/ECP-WarpX/impactx/pull/19/files#r778820245
```
0::Assertion `hi > lo' failed, file "/home/axel/src/impactx/build/_deps/fetchedamrex-src/Src/Base/AMReX_Algorithm.H", line 107, Msg: "Error - calling bisect but lo and hi don't describe a reasonable interval." !!!
```

I temporarily disabled calling it and enabled debug builds in CI (#43). The function probably needs to be fixed.

### PR DESCRIPTION
**Pull request by RemiLehe**
~~We should not resize the mesh when no particle is present.~~

This PR disables the Poisson solver when only one particle is present.

Fix #44

### CODE REVIEW

File: src/ImpactX.cpp [hunk 1]
```diff
@@ -139,8 +139,9 @@ namespace impactx
             // Note: The following operation assume that
             // the particles are in x, y, z coordinates.
             // Resize the mesh, based on `m_particle_container` extent
-            //   TODO: this is broken (fails in Debug mode)
-            //ResizeMesh();
+            if (m_particle_container.TotalNumberOfParticles(false,false) > 0) {
```
Comment (RemiLehe) [inline src/ImpactX.cpp]:
```suggestion
            if (m_particle_container->TotalNumberOfParticles(false,false) > 0) {
```

**Comment by RemiLehe** [comment]
Hm, the problem is actually that we have only 1 particle in this simulation.
In this case, the min and max position is equal, and therefore the resize fails.
@qianglbl In Impact, how do you resize the box when there is only 1 macroparticle in the beam? (granted: this is not a common case for production simulations, but it may be a useful case for testing.)

File: src/ImpactX.cpp [hunk 2]
```diff
@@ -139,8 +139,9 @@ namespace impactx
             // Note: The following operation assume that
             // the particles are in x, y, z coordinates.
             // Resize the mesh, based on `m_particle_container` extent
-            //   TODO: this is broken (fails in Debug mode)
-            //ResizeMesh();
+            if (m_particle_container->TotalNumberOfParticles(false,false) > 0) {
```
Comment (ax3l) [inline src/ImpactX.cpp]:
```suggestion
            if (m_particle_container->TotalNumberOfParticles(false,false) > 1) {
```

**Comment by ax3l** [comment]
@RemiLehe I think as you suggested, it makes sense to dynamically disable:
- grid scaling from `ResizeMesh`
- deposition
- Poisson solve
- space charge push

as soon as our particle container drops <2 particles.

All those components are not needed for a single particle sim :)

**Comment by ax3l** [comment]
@RemiLehe just a small hint: can you please fork the repo for future PRs? :)

**Review by ax3l — APPROVED** [review]
Thank you :)

**Pull request merged**